src/perf/lib/PerfServer.cpp

```diff
@@ -181,6 +225,86 @@ PerfServer::ConnectionCallback(
     return QUIC_STATUS_SUCCESS;
 }
 
+void
+PerfServer::IntroduceFixedDelay(uint32_t DelayUs)
+{
+    if (0 == DelayUs) {
+        return;
+    }
+
+    uint64_t Start = CxPlatTimeUs64();
+    while (CxPlatTimeDiff64(Start, CxPlatTimeUs64()) <= DelayUs) { /* busy wait*/ }
+}
+
+#ifndef _KERNEL_MODE
+
+#include <random>
+
+double
+PerfServer::CalculateVariableDelay(double DistributionParam)
+{
+    std::mt19937 random_generator(CxPlatTimeUs32());
+    std::exponential_distribution<> distribution(abs(DistributionParam));
+    return distribution(random_generator);
+}
+
+void
+PerfServer::IntroduceVariableDelay(uint32_t DelayUs)
+{
+    if (0 == DelayUs) {
+        return;
+    }
+
+    //
+    // Mean value of VariableDelay is expected to be DelayUs
+    //
+    double VariableDelay = CalculateVariableDelay(Lambda);
+
+    if ((VariableDelay + 1) < MaxFixedDelayUs) {
+        //
+        // Introduce a fixed delay up to a certain maximum value
+        //
+        IntroduceFixedDelay(static_cast<uint32_t>(VariableDelay));
+    } else {
+        //
+        // If the variable delay exceeds the maximum value,
+        // yield the thread for the max delay
+        //
+        CxPlatSleep(static_cast<uint32_t>(MaxFixedDelayUs/1000));
+    }
+}
+
+#else
+
+void
+PerfServer::IntroduceVariableDelay(uint32_t DelayUs)
+{
+    IntroduceFixedDelay(DelayUs);
+}
```

Comment:
I'd rather just not support variable delay at all on kernel for now, and update the switch in `SimulateDelay`